sys/net/crosslayer/ng_netreg/ng_netreg.c

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (C) 2015 Martine Lenders <mlenders@inf.fu-berlin.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file    ng_netreg.c
+ */
+
+#include <errno.h>
+
+#include "clist.h"
+#include "net/ng_netreg.h"
+#include "net/ng_nettype.h"
+#include "utlist.h"
+
+#define _INVALID_TYPE(type) (((type) == NG_NETTYPE_UNDEF) || ((type) >= NG_NETTYPE_NUMOF))
```

Comment:
Ah, I now get your second comment: The implementation somewhat depends on `NG_NETTYPE_UNDEF` being 0. If it is anything else, it must be changed anyways. But since it is the only value that is defined in `ng_nettype.h` I don't expect it to change.